Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -850,8 +913,12 @@ ExceptionOr<void> MediaSource::removeSourceBuffer(SourceBuffer& buffer)
 
             // 5.4 If the removed enabled audio track flag equals true, then queue a task to fire a simple event
             // named change at the HTMLMediaElement audioTracks list.
-            if (removedEnabledAudioTrack && m_mediaElement)
-                m_mediaElement->ensureAudioTracks().scheduleChangeEvent();
+            if (removedEnabledAudioTrack) {
+                ensureOnMainThread([weakMediaElement = m_mediaElement] {
```

Comment:
Since this is not a sync call, there is no guarantee that the media element is still attached to this media source.
We might want to add a check for that whenever we are doing this pattern.

Reply:
hmmm, we test the value of weakMediaElement here, if we the MediaSource is on the main thread, it is a sync call in practice